### PR TITLE
Add Apple keychain manager

### PR DESCRIPTION
**Pull request by sangonz**
This adds the `apple-keychain` manager with support for Apple's Keychain Access app.
It only imports passwords, not certificates or public/private keys.

### CODE REVIEW

**Comment by roddhjav** [comment]
Hi, thank for this PR.

Nice work with the apple key-chain weird text format. 
I will merge it, just allow me a bit of time to review it.

File: README.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ Pass import handles duplicates and is compatible with [browserpass][bp].
 | [1password][1password] | *See this [guide][export-1password]* | `pass import 1password file.csv` |
 | [1password4][1password] | *File > Export: CSV* | `pass import 1password4 file.csv` |
 | [1password4pif][1password] | *File > Export: 1PIF* | `pass import 1password4pif file.1pif` |
+| [apple-keychain][apple-keychain] | *See thie [guide][export-apple-keychain]* | `pass import apple-keychain file.txt` |
```
Comment (roddhjav) [inline README.md]: 's/thie/this/'

File: pass_import.py [hunk 2]
```diff
@@ -336,6 +337,212 @@ def clean(self, clean, convert):
...
+            ordered_entry[key] = entry.pop(key, None)
+
+        if self.all:
+            for key, value in entry.items():
+                ordered_entry[key] = value
+
+        self.data.append(ordered_entry)
+
+    def parse(self, file):
+        entry = {}
```
Comment (roddhjav) [inline pass_import.py]: The entry type is a `OrderedDict()`, not a `dict`

File: pass_import.py [hunk 3]
```diff
@@ -336,6 +337,212 @@ def clean(self, clean, convert):
...
+            if attr in human_keys:
+                key = human_keys[attr]
+
+            value = AppleKeychain._value_from_hextxt(entry['attributes'][attr])
+            if value:
+                passentry[key] = value
+
+        url = entry['attributes'].get('ptcl', {}).get('txt', '')
+        if url:
+            url = url.replace('htps', 'https')
```
Comment (roddhjav) [inline pass_import.py]: `url = url.replace('htps', 'https')` What is the purpose of this? There is no 'htps' in your test file

**Comment by roddhjav** [comment]
So your code looks good, I am going to merge after your fix the following things.

- To easily find the class, the importer class takes place in alphabetic order *after* the parent classes. So could you move the  `AppleKeychain` class between the `OnePassword4PIF` and `Bitwarden` classes.

- They seem to have reference of passowrd's group in your code. However, the test file you provided doe not seems to include group. Does this concept exist with apple-keychain?

**Comment by sangonz** [comment]
>     * To quickly find the class, the importer class takes place in alphabetic order _after_ the parent classes. So could you move the `AppleKeychain` class between the `OnePassword4PIF` and `Bitwarden` classes.

Ok, I _think_ I see that logic of parent classes, but shouldn't it be between `PasswordManagerPIF` and `OnePassword4PIF`?
```
# Parent classes
class PasswordManager():
class PasswordManagerCSV(PasswordManager):
class PasswordManagerXML(PasswordManager):
class PasswordManagerJSON(PasswordManager):
class PasswordManagerPIF(PasswordManagerJSON):
# Child classes
class AppleKeychain(PasswordManager):  # <--- here
class OnePassword4PIF(PasswordManagerPIF):
class OnePassword4(PasswordManagerCSV):
class OnePassword(PasswordManagerCSV):
class Bitwarden(PasswordManagerCSV):
```

>     * They seem to have references of password's group in your code. However, the test file you provided doe not seems to include group. Does this concept exist with apple-keychain?

Thanks for this very pertinent comment; the `desc` I was using is not for groupings. I took the time to find Apple's open source code and after some review I could see exactly the codes used. Since there is extremely few info in the internet about this I added the file [apple-keychain-security-internals.md ](https://gist.github.com/sangonz/601f4fd2f039d6ceb2198e2f9f4f01e0#apple-keychain-internals) to my gist.

I will leave only 'Notes", which follows Keychain Access' UI.

**Comment by roddhjav** [comment]
> but shouldn't it be between PasswordManagerPIF and OnePassword4PIF?

Yes, but the exact order is given by the `importers` global variable where OnePassword are named with a number; therefore, they take place before the letter.

File: pass_import.py (hunk 3 above)
Comment (sangonz) [reply pass_import.py]: It is the keyword used internally by Apple. See [protocol codes](https://gist.github.com/sangonz/601f4fd2f039d6ceb2198e2f9f4f01e0#protocol-codes). I've reproduced it in the test file and added more code substitutions.

**Comment by sangonz** [comment]
OK, I think I solved all issues.
Commit e5bc49c uniquely moves `AppleKeychain` class code to the correct place, to make reviewing code easier.

**Comment by roddhjav** [comment]
Good work! merged. Thank again for this very nice PR.

**Pull request merged**